### DISCUSSION TITLE
how to test useController using jest?

### DISCUSSION THREAD

**Discussion by ednilsoncs** [react-hook-form/react-hook-form#4243] [General] [answered]
hello, I was here trying to test a react component using jest. I ended up with a situation very similar to the one that was in the documentation.

How do I test this <Input /> component and mock the control parameter?

`import React from "react";
import { TextField } from "@material-ui/core";
import { useController, useForm } from "react-hook-form";

function Input({ control, name }) {
  const {
    field: { ref, ...inputProps },
    meta: { invalid, isTouched, isDirty },
  } = useController({
    name,
    control,
    rules: { required: true },
    defaultValue: "",
  });

  return <TextField {...inputProps} inputRef={ref} />;
}

function App() {
  const { control } = useForm();
  
  return <Input name="firstName" control={control} />;
}
`

**Comment by bluebill1049** [comment, marked as answer]
have a look at this file: https://github.com/react-hook-form/react-hook-form/blob/v7/src/__tests__/useController.test.tsx

**Reply by ednilsoncs** [reply]
Thank you very much, I will check if it works for me

**Reply by sanatg7582** [reply]
Url is not working

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/blob/master/src/__tests__/useController.test.tsx

**Reply by sanatg7582** [reply]
I didn't get any solution in the URL for the above question Please help me how can i write test cases for Input component in the component we are using useController hooks from react-form-hook
